### PR TITLE
Add WOQ GPTQ frontend and example

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change

GPTQ WOQ support with llmruntime=False.

## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by changwangss** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1400/

File: tests/CI/test_quantization.py [hunk 1]
```diff
@@ -421,8 +405,27 @@ def test_quantization_for_llm(self):
                                                      device_map="cpu"
                                                 )
         output = bit8_model(dummy_input)
-        self.assertTrue(isclose(float(output[0][0][0][0]), -7.2695, rel_tol=1e-04))
-
+        self.assertTrue(isclose(float(output[0][0][0][0]), 0.1674591302871704, rel_tol=1e-04))
+
+        #GPTQ
```
Comment (a32543254) [inline tests/CI/test_quantization.py:410]: could we also add awq test?

File: intel_extension_for_transformers/llm/quantization/nn/modules.py [hunk 2]
```diff
@@ -94,7 +97,7 @@ def __init__(
         bias=True,
         compute_dtype="fp32",
         compress_statistics=True,
-        weight_dtype="int4_fullrange",
+        weight_dtype="int4_clip",
```
Comment (a32543254) [inline intel_extension_for_transformers/llm/quantization/nn/modules.py:100]: why change int4_fullrange to int4_clip

**Review by a32543254 — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/llm/quantization/nn/modules.py (hunk 2 above)
Comment (changwangss) [reply intel_extension_for_transformers/llm/quantization/nn/modules.py:100]: int4_fullrange is not a general format, int4_clip is more popular.

File: tests/CI/test_quantization.py (hunk 1 above)
Comment (changwangss) [reply tests/CI/test_quantization.py:410]: AWQ, We have supported it before，but it is generally replace all linear from torch.nn.Linear to QuantizedLinearQbits

**Comment by changwangss** [comment]
local test facebook/opt-125m
fp32 accuracy:
```
numactl -m 1 -C 56-111 python run_generation.py --model facebook/opt-125m  --accuracy --batch_size 56  
```
```
|     Task     |Version|Metric| Value |   |Stderr|
|--------------|------:|------|------:|---|-----:|
|lambada_openai|      0|ppl   |26.0209|±  |0.9419|
|              |       |acc   | 0.3790|±  |0.0068|

```
int8 accuracy:
```
numactl -m 1 -C 56-111 python run_generation.py --model facebook/opt-125m --woq --wo
q_weight_dtype "int4_clip" --woq_compute_dtype "fp32" --woq_algo "GPTQ"  --gptq_block_size 32  --gptq_actorder --accuracy --batch_size 56  
```
```
|     Task     |Version|Metric| Value |   |Stderr|
|--------------|------:|------|------:|---|-----:|
|lambada_openai|      0|ppl   |31.1117|±  |1.1767|
|              |       |acc   | 0.3526|±  |0.0067|
```

**Comment by changwangss** [comment]
extension for llama2: https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1402/

**Comment by changwangss** [comment]
https://inteltf-jenk.sh.intel.com/job/nlp_toolkit_optimize_validation_localtest/4115/

File: intel_extension_for_transformers/llm/quantization/nn/modules.py [hunk 3]
```diff
@@ -134,18 +145,100 @@ def forward(self, x: torch.Tensor):
...
+        if q_config.gptq_quantize_config["sym"]:
+            gptq_zeros = torch.empty(0, dtype=torch.int8)
+
+        if not q_config.gptq_quantize_config["desc_act"]:
+            g_idx = torch.empty(0, dtype=torch.int32)
+
+        packw = torch.ops.bestlaop.woq_packq(
+            int_weight.contiguous(),
+            gptq_scales.float().contiguous(),
+            gptq_zeros,
```
Comment (zhewang1-intc) [inline intel_extension_for_transformers/llm/quantization/nn/modules.py:220]: emm, i guess all tensors passed to `bestlaop.packq` should be contiguous, incuding zp & g_idx

Comment (changwangss) [reply intel_extension_for_transformers/llm/quantization/nn/modules.py]: fix :)

**Comment by changwangss** [comment]
llama: https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1409/
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1408/

**Comment by changwangss** [comment]
> > llama: https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1409/ https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1408/
> 
> lm-eval?

sure
